apps/fhir/bluebutton/views/generic.py

```diff
@@ -87,6 +91,16 @@ def initial(self, request, resource_type, *args, **kwargs):
         logger.debug("resource_type: %s" % resource_type)
         logger.debug("Interaction: read")
         logger.debug("Request.path: %s" % request.path)
+        req_meta = request.META
+
+        if "HTTP_AUTHORIZATION" in req_meta:
+            access_token = req_meta["HTTP_AUTHORIZATION"].split(" ")[1]
+            try:
+                at = AccessToken.objects.get(token=access_token)
+                logger.info(f"ACCESS_TOKEN_ID: {at.application.id}")
+                logger.info(f"ACCESS_TOKEN_HASH: {hashlib.sha256(str(access_token).encode('utf-8')).hexdigest()}")
```

Comment:
The ticket says we want a Access Token User ID and User Hash, but this seems to be the application id. Do we have access to the user id here instead? As it is, it's unclear how I would use these to tie the logs to a specific user.

Reply:
It's a pattern taken from request_logging.py where access tokens are similarly extracted for logging purposes. Our use case may be slightly different. I swapped this out for at.id here.

Reply:
It seems like the `at.id` is showing the user id associated with the application, not necessarily the one associated with the access token. I might recommend starting a fresh server, and generating an access token, looking in Django Admin to see the details of the access token, then setting a breakpoint at this point in the code to see if the access token details are visible anywhere there. As far as I understand, it shouldn't matter exactly what field is logged, as long as it's something that we can search on in Django admin to find the user, and I'm still not seeing that in my testing still.

Reply:
That makes sense! I can walk that back into a value searchable from django admin. I had been approaching it to match up with the similar AT logging done in the HHS Oauth Server request audit logging, but the values we need may be even easier.